lib/apiservers/engine/backends/system.go

```diff
@@ -15,31 +15,148 @@
 package vicbackends
 
 import (
+	"bufio"
 	"fmt"
+	"os"
+	"regexp"
 	"runtime"
+	"strings"
+	"time"
 
 	"golang.org/x/net/context"
 
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/vic/lib/apiservers/portlayer/client"
+	"github.com/vmware/vic/lib/apiservers/portlayer/client/misc"
+
+	"github.com/docker/docker/pkg/parsers/kernel"
+	"github.com/docker/docker/pkg/platform"
+	"github.com/docker/docker/pkg/system"
 	"github.com/docker/engine-api/types"
 	"github.com/docker/engine-api/types/events"
 	"github.com/docker/engine-api/types/filters"
+	"github.com/docker/go-connections/sockets"
 )
 
 type System struct {
 	ProductName string
 }
 
+type ContainerStatus struct {
+	count      int
+	numRunning int
+	numStopped int
+	numPaused  int
+}
+
+var (
+	etcReleaseFile    = "/etc/os-release"
+	usrlibReleaseFile = "/usr/lib/os-release"
+)
+
 func (s *System) SystemInfo() (*types.Info, error) {
-	Driver := "Portlayer Storage"
-	IndexServerAddress := "https://index.docker.io/v1/"
-	ServerVersion := "0.0.1"
-	Name := "VIC"
+	// Set VIC specific data
+	Driver := s.ProductName + " Backend Engine"
+	IndexServerAddress := "registry-1.docker.io"
+	ServerVersion := "0.3"
+
+	// Use docker pkgs to get some system data
+	kernelVersion := "<unknown>"
+	if kv, err := kernel.GetKernelVersion(); err != nil {
+		log.Warnf("Could not get kernel version: %v", err)
+	} else {
+		kernelVersion = kv.String()
+	}
+
+	operatingSystem := getOperatingSystem()
+
+	meminfo, err := system.ReadMemInfo()
+	if err != nil {
+		log.Errorf("Could not read system memory info: %v", err)
+	}
 
+	// Check if portlayer server is up
+	plClient := PortLayerClient()
+
+	systemStatus := make([][2]string, 1)
+	systemStatus[0][0] = s.ProductName + " Backend Engine"
+	if pingPortlayer(plClient) {
+		systemStatus[0][1] = "RUNNING"
+	} else {
+		systemStatus[0][1] = "STOPPED"
+	}
+
+	// Retrieve info about the VCH from Exec port layer
+	//FIXME: Write the portlayer code to do this
```

Comment:
We could get the info ourselves, but this is the personality server.  Should it even know about any VMW specific APIs?